fvm/src/gas/mod.rs

```diff
@@ -152,53 +153,52 @@ impl Mul<i32> for Gas {
 
 pub struct GasTracker {
     gas_limit: Gas,
-    gas_used: Gas,
-    trace: Option<Vec<GasCharge>>,
+    gas_used: Cell<Gas>,
+    trace: Option<RefCell<Vec<GasCharge>>>,
```

Comment:
Putting the option on the outside means we don't have to pay for the dynamic refcell check unless we're actually tracing. Ideally we'd just use a cell, but `Cell::update` is still unstable and this isn't performance critical.